tests/unit/style/properties/serialization.rs

```diff
@@ -1013,4 +1013,39 @@ mod shorthand_serialization {
             );
         }
     }
+
+    mod scroll_snap_type {
+        pub use super::*;
+        use style::properties::longhands::scroll_snap_type_x::computed_value::T as ScrollSnapTypeXValue;
+        use style::properties::longhands::scroll_snap_type_y::computed_value::T as ScrollSnapTypeYValue;
+
+        #[test]
+        fn should_serialize_to_empty_string_if_sub_types_not_equal() {
+            let mut properties = Vec::new();
+
+            let none_x = DeclaredValue::Value(ScrollSnapTypeXValue::none);
+            properties.push(PropertyDeclaration::ScrollSnapTypeX(none_x));
+
+            let mandatory_y = DeclaredValue::Value(ScrollSnapTypeYValue::mandatory);
+            properties.push(PropertyDeclaration::ScrollSnapTypeY(mandatory_y));
+
+            let serialization = shorthand_properties_to_string(properties);
```

Comment:
I still had to use `assert_eq!(string, "scroll-snap-type: ;");` even after using `single_value_to_css`.